int16 reward
string behaviour
---
int16 success